File: Java/Algorithms/String/Swap.java
/**
 * Class Swap
 */
public class Swap {
    /**
     * Swaps the characters of a given string
     *
     * @param str a given string
     * @param i the index of the first character
     * @param j the index of the second character
     * @return String
     */
    public static String swa(String str, int i, int j) {
        return str.substring(0, i) +
               str.charAt(j) +
               str.substring(i + 1, j) +
               str.charAt(i) +
               str.substring(j + 1);
    }
}

File: C#/Algorithms/String/Swap.cs
/**
 * Class Swap
 */
public class Swap {
    /**
     * Swaps the characters of a given string
     *
     * @param string str
     * @param int i 
     * @param int j 
     * @return string
     */
    public string swap(string str, int i, int j) {
        return str.Substring(0, i) +
               str[j] +
               str.Substring(i + 1, j - i - 1) +
               str[i] +
               str.Substring(j + 1);
    }
}